File: src/main/java/vip/mystery0/base/springboot/config/WebLogAspect.java
package vip.mystery0.base.springboot.config;

import org.aspectj.lang.ProceedingJoinPoint;
import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;
import vip.mystery0.base.springboot.utils.trace.TraceHelper;
import vip.mystery0.base.springboot.utils.trace.TraceLogUtil;

import javax.servlet.http.HttpServletRequest;
import java.util.Objects;

public class WebLogAspect {
    public static void doWebLog(ProceedingJoinPoint joinPoint, int maxLength) throws Throwable {
        ServletRequestAttributes attributes = (ServletRequestAttributes) RequestContextHolder.getRequestAttributes();
        HttpServletRequest request = Objects.requireNonNull(attributes).getRequest();
        TraceHelper.beginTrace(request);
        TraceLogUtil.logRequest(request, maxLength);
        Object result = joinPoint.proceed();
        TraceLogUtil.logResponse(result);
        TraceHelper.endTrace();
    }
}

File: src/main/java/vip/mystery0/base/springboot/config/WebLogAspect.kt
package vip.mystery0.base.springboot.config

import org.aspectj.lang.ProceedingJoinPoint
import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes
import vip.mystery0.base.springboot.utils.trace.TraceHelper
import vip.mystery0.base.springboot.utils.trace.TraceLogUtil
import vip.mystery0.base.springboot.utils.trace.TraceLogUtil.log
import vip.mystery0.tools.kotlin.utils.doByTry
import java.util.*

object WebLogAspect {
    fun log(joinPoint: ProceedingJoinPoint): Any? = doWebLog(joinPoint)
}

fun doWebLog(joinPoint: ProceedingJoinPoint): Any? {
    val attributes = RequestContextHolder.getRequestAttributes() as ServletRequestAttributes?
    val request = Objects.requireNonNull(attributes)!!.request
    TraceHelper.beginTrace(request)
    doByTry { TraceLogUtil.logRequest(request, joinPoint) }.log()
    val result = joinPoint.proceed()
    doByTry { TraceLogUtil.logResponse(result) }.log()
    TraceHelper.endTrace()
    return result
}